modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/InterpolationQuantileMethod.java

```diff
@@ -6,75 +6,114 @@
 package com.opengamma.strata.math.impl.statistics.descriptive;
 
 import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.DoubleArrayMath;
 import com.opengamma.strata.collect.array.DoubleArray;
 
 /**
  * Implementation of a quantile estimator.
  * <p>
- * The quantile is linearly interpolated between two sample values. The probability dimension <i>p<subscript>i</subscript></i> 
- * on which the interpolation take place (X axis) varies between actual implementation of the abstract class. 
- * For each probability <i>p<subscript>i</subscript></i>, the cumulative distribution value is the sample value with 
- * same index. The index used above are the Java index plus 1.
+ * The quantile is linearly interpolated between two sample values. The probability dimension
+ * <i>p<subscript>i</subscript> on which the interpolation take place (X axis) varies between actual implementation
+ * of the abstract class. For each probability <i>p<subscript>i</subscript></i>, the cumulative distribution value is
+ * the sample value with same index. The index used above are the Java index plus 1.
  * <p>
  * Reference: Value-At-Risk, OpenGamma Documentation 31, Version 0.1, April 2015.
  */
+
 public abstract class InterpolationQuantileMethod
     extends QuantileCalculationMethod {
 
   @Override
-  protected double quantile(double level, DoubleArray sortedSample, boolean isExtrapolated) {
+  protected QuantileResult quantile(double level, DoubleArray sample, boolean isExtrapolated) {
     ArgChecker.isTrue(level > 0, "Quantile should be above 0.");
     ArgChecker.isTrue(level < 1, "Quantile should be below 1.");
-    int sampleSize = sortedSample.size();
-    double adjustedLevel =
-        checkIndex(level * sampleCorrection(sampleSize) + indexCorrection(), sortedSample.size(), isExtrapolated);
+    int sampleSize = sampleCorrection(sample.size());
+    double adjustedLevel = checkIndex(level * sampleSize + indexCorrection(), sample.size(), isExtrapolated);
+    double[] order = createIndexArray(sample.size());
+    double[] s = sample.toArray();
+    DoubleArrayMath.sortPairs(s, order);
     int lowerIndex = (int) Math.floor(adjustedLevel);
     int upperIndex = (int) Math.ceil(adjustedLevel);
     double lowerWeight = upperIndex - adjustedLevel;
     double upperWeight = 1d - lowerWeight;
-    return lowerWeight * sortedSample.get(lowerIndex - 1) + upperWeight * sortedSample.get(upperIndex - 1);
+    return QuantileResult.of(
+        lowerWeight * s[lowerIndex - 1] + upperWeight * s[upperIndex - 1],
+        new int[]{(int) order[(int) Math.floor(adjustedLevel) - 1], (int) order[(int) Math.ceil(adjustedLevel) - 1]},
```

Comment:
`(int) Math.floor(adjustedLevel)` and `(int) Math.ceil(adjustedLevel)` are already computed in 35 and 36, respectively.